pulsar-client-cpp/include/pulsar/MessageId.h

```diff
@@ -44,6 +45,12 @@ class PULSAR_PUBLIC MessageId {
      */
     explicit MessageId(int32_t partition, int64_t ledgerId, int64_t entryId, int32_t batchIndex);
 
+    /**
+     * Construct the ChunkMessageId with first and last message id's param
+     */
+    explicit MessageId(int32_t firstPartition, int64_t firstLedgerId, int64_t firstEntryId, int32_t firstBatchIndex,
+                     int32_t lastPartition, int64_t lastLedgerId, int64_t lastEntryId, int32_t lastBatchIndex);
```

Comment:
The code readability of constructor with 8 parameters is bad. (Though I think the current constructor with 4 parameters is also not good).

My suggestion is like

```c++
/**
 * Construct a chunked message ID.
 */
MessageId(const MessageId& firstMessageId, const MessageId& lastMessageId);
```

Reply:
OK, I'll revise it. In fact, I used to be this kind of design, but I referred to the above example（(4 parameters) and changed it to the current one.HHH

Reply:
IMO, a better solution is adding a builder class and keep the current constructor for backward compatibility.

```c++
// A simple MessageId, which is equivalent with Java's BatchMessageIdImpl
const auto msgId1 = MessageIdBuilder()
    .ledgerId(0L)
    .entryId(0L)
    .batchIndex(1)
    .build(); // the default partition index is -1
// A chunked MessageId, which is equivalent with Java's ChunkMessageIdImpl
const auto msgId2 = MessageIdBuilder()
    .ledgerId(0L)
    .entryId(0L)
    .batchIndex(1)
    .firstChunkMessageId(/* ... */)
    .build();
```

But it will involve many changes. So I think in this PR, adding a constructor that accepts two `MessageId` parameters is enough.